# origin: Debian, #211717
# cset: n/a
# inclusion: not submitted
# description: Do not unshare files for pid == 1 in kernel/fork.c 
# revision date: 2004-09-03

--- kernel-source-2.4.26/kernel/fork.c	2004-04-14 23:05:40.000000000 +1000
+++ kernel-source-2.4.26-1/kernel/fork.c	2004-04-17 14:24:07.000000000 +1000
@@ -562,7 +562,7 @@
 		
 	/* This can race but the race causes us to copy when we don't
 	   need to and drop the copy */
-	if(atomic_read(&files->count) == 1)
+	if(atomic_read(&files->count) == 1 || current->pid == 1)
 	{
 		atomic_inc(&files->count);
 		return 0;
